### DISCUSSION TITLE
Clarity on syntax especially as regards to line endings

### DISCUSSION THREAD

**Discussion by alizzycraft** [hylo-lang/hylo#824] [Language design]
Regarding the fact that Val doesn't use line endings, apart from Js/Ts the semicolon is usually required when there are also braces {}, I guess I'm used to being able to ignore whitespace mentally via explicit endings of blocks and lines, not just blocks. It looks like Val has gone with mapping the new line character to line endings, if I'm not mistaken, which means we do need to care about whitespace using it?

I'm interested in the reasoning behind this design choice and how it aligns with the rest of Val's syntax. Hopefully getting an understanding of the thought process would be beneficial for others too.

**Comment by kyouko-taiga** [comment]
We've based our syntax on Swift, which made the same choice. In general, starting from an existing language gives us confidence in specific design choices that have been shown to work. The fact that Javascript, Typescript, and Kotlin (to cite a few) have gone the same road is just further evidence that we're not making a mistake.

Semicolons add unnecessary noise in most cases because of the natural way statements read. There might be useful sometimes, which is why we can add them optionally.
